telegraf/tests/test.sh

```diff
@@ -1,27 +1,31 @@
 #!/bin/sh
+#/ Usage: test.sh <pkg_ident>
+#/
+#/ Example: test.sh core/php/7.2.8/20181108151533
+#/
 
-TESTDIR="$(dirname "${0}")"
-PLANDIR="$(dirname "${TESTDIR}")"
-SKIPBUILD=${SKIPBUILD:-0}
+set -euo pipefail
 
-hab pkg install core/bats --binlink
+source "$(dirname "${0}")/../../bin/ci/test_helpers.sh"
 
-if [ "${SKIPBUILD}" -eq 0 ]; then
-  source "${PLANDIR}/plan.sh"
-  # Unload the service if its already loaded.
-  hab svc unload "${HAB_ORIGIN}/${pkg_name}"
-
-  set -e
-  pushd "${PLANDIR}" > /dev/null
-  build
-  source results/last_build.env
-  hab pkg install "results/${pkg_artifact}" --binlink --force
-  hab svc load "${pkg_ident}"
-  popd > /dev/null
-  set +e
-
-  # Wait for service to start
-  sleep 10
+if [[ -z "${1:-}" ]]; then
+  grep '^#/' < "${0}" | cut -c4-
+	exit 1
 fi
 
-bats "${TESTDIR}/test.bats"
+TEST_PKG_IDENT="${1}"
+export TEST_PKG_IDENT
+hab pkg install core/bats --binlink
+hab pkg install "${TEST_PKG_IDENT}"
+
+ci_ensure_supervisor_running
+ci_load_service "${TEST_PKG_IDENT}"
+
+# Allow service start
```

Comment:
ci_load_service takes an optional second parameter  (default 5 seconds)  to tell it how long to wait for the service to start, so this sleep shouldn't be necessary. If it is, I'd love to understand why so we can improve our test helpers :smile:

Reply:
@smacfarlane `ci_load_service` has a timeout for service to be loaded. It will at most, wait that time. If the service loads before that time is up, it will exit the function cleanly.

What it is *not* checking for is a service to be **ready**. If something is a network listener, and slow (i'm looking at you, java) it can take some time after the service is loaded, for it to be considered ready for tests to talk to the service.

Thats what this sleep is for.